packages/ember-data/lib/system/store.js

```diff
@@ -1073,6 +1073,7 @@ Store = Service.extend({
     }
 
     typeMap.findAllCache = null;
+    this.typeMapFor(type).metadata = Ember.create(null);
```

Comment:
You already have `typeMap` available here, another call to `typeMapFor()` should not be needed.